<?xml version="1.0"?>
<launch>
    <arg name="cloud_topic" default=""/>
    <arg name="processed_cloud_topic" default="processed_cloud"/>
    <arg name="obstacle_cloud_topic" default="obstacle_cloud"/>
    <arg name="target_frame" default="/base_link" />
    <arg name="obstacles_detection_topic" default="/detected_obstacles"/>
    <arg name="obstacle_detection_config_file"
         default="$(find mas_perception_libs)/ros/config/obstacle_detection_default_configs.yaml" />

    <node pkg="mas_perception_libs" type="cloud_obstacle_detection" name="cloud_obstacle_detection" output="screen"
          ns="mas_perception">
        <rosparam file="$(arg obstacle_detection_config_file)" command="load"/>
        <param name="cloud_topic" type="string" value="$(arg cloud_topic)" />
        <param name="processed_cloud_topic" type="string" value="$(arg processed_cloud_topic)" />
        <param name="obstacle_cloud_topic" type="string" value="$(arg obstacle_cloud_topic)" />
        <param name="obstacles_detection_topic" type="string" value="$(arg obstacles_detection_topic)" />
        <param name="target_frame" type="string" value="$(arg target_frame)" />
    </node>

    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan_node" output="screen"
          ns="mas_perception">
        <remap from="cloud_in" to="cloud_obstacle_detection/$(arg obstacle_cloud_topic)" />
        <remap from="scan" to="obstacle_scan" />
    </node>
</launch>
